project/dexter-vs/Dexter/Core/UI/Settings/SettingsControl.cs

```diff
@@ -198,9 +198,42 @@ DexterInfo IDexterInfoProvider.Load()
             return dexterInfo;
         }
 
+        private void createUser()
+        {
+            // We need to save current url 
+            DexterInfo dexterInfo = GetDexterInfoFromSettings();
+            var result = "";
+
+            if (!validator.ValidateServerConnection(dexterInfo, out result))
+            {
+                MessageBox.Show("Couldn't connect to Dexter server. Setting analysis mode to standalone", "Dexter warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+                standaloneCheckBox.Checked = true;
+                dexterInfo.standalone = true;
+                Save(dexterInfo);
+            }
+            else if (validator.ValidateUserCredentials(dexterInfo, out result))
+            {
+                MessageBox.Show($"Couldn't create user: user {dexterInfo.userName} already exists", "Dexter warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+            }
+            else
+            {
+                DexterClient.Instance = new DexterClient(new DexterHttpClientWrapper(dexterInfoProvider), dexterInfoProvider);
```

Comment:
Yes, of course. I have several arguments. 

1. **There is no need to use singletons**

Typical implementation of REST API client is to create separate instance of client **each time it is needed**, without using singletons. Just take a look at typical implementations available on the web: 

- [code 1](https://www.codeproject.com/Tips/497123/How-to-make-REST-requests-with-Csharp)
- [code 2](https://stackoverflow.com/questions/9620278/how-do-i-make-calls-to-a-rest-api-using-c)

Singleton instance of REST API client makes sense in several cases, e.g. when we need to maintain connection through whole application liftetime. But this is not the case. 

2. **Implementation issues**

Singleton implementation of Dexter Client is broken for several reasons. Constructor should not be publicly exposed. Instance creation should be performed in getter. Moreover, it is being used in a way which replaces previous instance:

`DexterClient.Instance = new DexterClient(new DexterHttpClientWrapper(dexterInfoProvider), dexterInfoProvider);`
				
3. **Implementation issues cont.**

Even if this singleton creation would be OK, we are still, in fact, not using singletons. Consider this code from DexterHttpClientWrapper:

        public Task<HttpResponseMessage> GetAsync(string requestUri)
        {
            HttpClient httpClient = new HttpClient();
            setRequestHeader(httpClient);
            return httpClient.GetAsync(requestUri);
        }

As you can see, each time a GetAsync is called, a new HttpClient is being created. So each time we are creating new instance, even though we are using the same instance of DexterClient. And all methods in DexterHttpClientWrapper are implemented this way. 

3. **Singletons are antipattern**

Even though Singletons were originally one of design patterns invented by Gang of Four, there are often considered antipattern nowadays. They are overused, hard to test, they violate OOP and are error-prone. 

- [source 1](http://archive.oreilly.com/cs/user/view/cs_msg/23417) 
- [source 2](https://stackoverflow.com/questions/137975/what-is-so-bad-about-singletons)

To cite one of the authors: 

> A singleton, as defined by the GoF, is a data structure which:
> 
> - Grants global access to an object, and
> - Enforces that only one instance of the object can ever exist.
> The first is generally considered a bad thing. We don't like globals. The second is a bit more subtle, but generally, there are virtually no cases where this is a reasonable restriction to enforce.
> 

So I would strongly advise to avoid them.